### PR TITLE
Deprecation message in PHP 8.1, clearer code

### PR DESCRIPTION
**Pull request by MacJoom**
### Summary of Changes
- Avoid deprecation message in PHP 8.1
- clearer code
### Testing Instructions
Setup a custom module
Setup an article with a {loadmodule costum}

Turn on all PHP Warnings
display_errors = On
error_reporting = E_ALL | E_STRICT

### Actual result BEFORE applying this Pull Request

Deprecated: trim(): Passing null to parameter #1 ($string) of type string is deprecated in /var/www/clients/client2/web14/web/joomla-cms/plugins/content/loadmodule/loadmodule.php on line 115

### Expected result AFTER applying this Pull Request

No warning anymore

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: plugins/content/loadmodule/loadmodule.php [hunk 1]
```diff
@@ -101,22 +101,20 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
...
-                // We may not have a specific module so set to null
-                if (!array_key_exists(1, $matchesmodlist)) {
-                    $matchesmodlist[1] = null;
+                //first parameter is the module, will be prefixed with mod_ later
+                $module = trim($matchesmodlist[0]);
+                // Second parameter is the title
+                if (array_key_exists(1, $matchesmodlist)) {
+                    $title   = htmlspecialchars_decode(trim($matchesmodlist[1]));
+                } else {
+                    $title   = "";
```
Comment (richard67) [inline plugins/content/loadmodule/loadmodule.php]: Please use single quotes and not double quotes here. We use double quotes only when necessary or helpful, e.g. when having a „\n“ in the string. But here we don’t have that so we should use single quotes.

Comment (MacJoom) [reply plugins/content/loadmodule/loadmodule.php]: OK - understood - will change this

File: plugins/content/loadmodule/loadmodule.php [hunk 2]
```diff
@@ -101,22 +101,20 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
...
-                // We may not have a specific module so set to null
-                if (!array_key_exists(1, $matchesmodlist)) {
-                    $matchesmodlist[1] = null;
+                //first parameter is the module, will be prefixed with mod_ later
+                $module = trim($matchesmodlist[0]);
+                // Second parameter is the title
+                if (array_key_exists(1, $matchesmodlist)) {
+                    $title   = htmlspecialchars_decode(trim($matchesmodlist[1]));
+                } else {
+                    $title   = '';
```
Comment (Quy) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion
                // First parameter is the module, will be prefixed with mod_ later
                $module = trim($matchesmodlist[0]);
                // Second parameter is the title
                if (array_key_exists(1, $matchesmodlist)) {
                    $title = htmlspecialchars_decode(trim($matchesmodlist[1]));
                } else {
                    $title = '';
```

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 992ef0522530f329467da90219a7606bcf64121d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39003">issues.joomla.org/tracker/joomla-cms/39003</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 992ef0522530f329467da90219a7606bcf64121d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39003">issues.joomla.org/tracker/joomla-cms/39003</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39003">issues.joomla.org/tracker/joomla-cms/39003</a>.</sub>

File: plugins/content/loadmodule/loadmodule.php [hunk 3]
```diff
@@ -101,22 +101,20 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
...
-                if (!array_key_exists(1, $matchesmodlist)) {
-                    $matchesmodlist[1] = null;
+                // First parameter is the module, will be prefixed with mod_ later
+                $module = trim($matchesmodlist[0]);
+                // Second parameter is the title
+                if (array_key_exists(1, $matchesmodlist)) {
+                    $title = htmlspecialchars_decode(trim($matchesmodlist[1]));
+                } else {
+                    $title = '';
                 }
```
Comment (HLeithner) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion
                // First parameter is the module, will be prefixed with mod_ later
                $module = trim($matchesmodlist[0]);
                $title = '';

                // Second parameter is the title
                if (array_key_exists(1, $matchesmodlist)) {
                    $title = htmlspecialchars_decode(trim($matchesmodlist[1]));
                }

```

File: plugins/content/loadmodule/loadmodule.php [hunk 4]
```diff
@@ -101,22 +101,20 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
...
-
-                // We may not have a module style so fall back to the plugin default.
-                if (!array_key_exists(2, $matchesmodlist)) {
-                    $matchesmodlist[2] = $stylemod;
+                // Third paramater is the module style, (fallback is the plugin default set earlier).
+                if (array_key_exists(2, $matchesmodlist)) {
+                    $stylemod = trim($matchesmodlist[2]);
                 }
```
Comment (HLeithner) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion
                $stylemod = '';
                if (array_key_exists(2, $matchesmodlist)) {
                    $stylemod = trim($matchesmodlist[2]);
                }
```

**Review by HLeithner — CHANGES_REQUESTED** [review]
We try to avoid else where easily possible.

**Comment by bembelimen** [comment]
Remove RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39003">issues.joomla.org/tracker/joomla-cms/39003</a>.</sub>

File: plugins/content/loadmodule/loadmodule.php [hunk 5]
```diff
@@ -101,22 +101,21 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
...
-                if (!array_key_exists(1, $matchesmodlist)) {
-                    $matchesmodlist[1] = null;
+                // First parameter is the module, will be prefixed with mod_ later
+                $module = trim($matchesmodlist[0]);
+                // Second parameter is the title
+                if (array_key_exists(1, $matchesmodlist)) {
+                    $title = htmlspecialchars_decode(trim($matchesmodlist[1]));
+                } else {
+                    $title = '';
                 }
```
Comment (Quy) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion
                $title = '';
                if (array_key_exists(1, $matchesmodlist)) {
                    $title = htmlspecialchars_decode(trim($matchesmodlist[1]));
                }
```

**Comment by laoneo** [comment]
@HLeithner can you approve when the code is ok, then people can test it again?

File: plugins/content/loadmodule/loadmodule.php [hunk 6]
```diff
@@ -106,22 +106,20 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
...
+                // Second parameter is the title
+                $title = '';
+                if (array_key_exists(1, $matchesmodlist)) {
+                    $title = htmlspecialchars_decode(trim($matchesmodlist[1]));
                 }
-
-                // We may not have a module style so fall back to the plugin default.
-                if (!array_key_exists(2, $matchesmodlist)) {
-                    $matchesmodlist[2] = $stylemod;
+                // Third paramater is the module style, (fallback is the plugin default set earlier).
```
Comment (HLeithner) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion

                // Third paramater is the module style, (fallback is the plugin default set earlier).
```

File: plugins/content/loadmodule/loadmodule.php [hunk 7]
```diff
@@ -106,22 +106,20 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
             foreach ($matchesmod as $matchmod) {
                 $matchesmodlist = explode(',', $matchmod[1]);
 
-                // We may not have a specific module so set to null
-                if (!array_key_exists(1, $matchesmodlist)) {
-                    $matchesmodlist[1] = null;
+                // First parameter is the module, will be prefixed with mod_ later
+                $module = trim($matchesmodlist[0]);
+                // Second parameter is the title
```
Comment (HLeithner) [inline plugins/content/loadmodule/loadmodule.php:112]:
```suggestion

                // Second parameter is the title
```

File: plugins/content/loadmodule/loadmodule.php [hunk 8]
```diff
@@ -106,22 +106,20 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
...
+                $title = '';
+                if (array_key_exists(1, $matchesmodlist)) {
```
Comment (HLeithner) [inline plugins/content/loadmodule/loadmodule.php:115]:
```suggestion

                if (array_key_exists(1, $matchesmodlist)) {
```

**Review by HLeithner — CHANGES_REQUESTED** [review]

File: plugins/content/loadmodule/loadmodule.php [hunk 9]
```diff
@@ -106,22 +106,22 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
             foreach ($matchesmod as $matchmod) {
                 $matchesmodlist = explode(',', $matchmod[1]);
 
-                // We may not have a specific module so set to null
-                if (!array_key_exists(1, $matchesmodlist)) {
-                    $matchesmodlist[1] = null;
-                }
+                // First parameter is the module, will be prefixed with mod_ later
+                $module = trim($matchesmodlist[0]);
+                // Second parameter is the title
```
Comment (HLeithner) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion
                $module = trim($matchesmodlist[0]);

                // Second parameter is the title
```

File: plugins/content/loadmodule/loadmodule.php [hunk 10]
```diff
@@ -106,22 +106,22 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
...
+                if (array_key_exists(1, $matchesmodlist)) {
+                    $title = htmlspecialchars_decode(trim($matchesmodlist[1]));
                 }
 
-                $module = trim($matchesmodlist[0]);
-                $name   = htmlspecialchars_decode(trim($matchesmodlist[1]));
-                $stylemod  = trim($matchesmodlist[2]);
+                // Third paramater is the module style, (fallback is the plugin default set earlier).
+                $stylemod = '';
+                if (array_key_exists(2, $matchesmodlist)) {
```
Comment (HLeithner) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion
                $stylemod = '';

                if (array_key_exists(2, $matchesmodlist)) {
```

**Review by HLeithner — CHANGES_REQUESTED** [review]

**Review by HLeithner — APPROVED** [review]
I added some blank lines which we have normally before comments and if statements. Rest looks good for me now.

**Comment by carlitorweb** [comment]
I have tested this item :white_check_mark: successfully on 6b93585871e98a1baca8fe82e4b6be70f91d2245<br /><br />**Just a note:** In 8.2 do not show this deprecated message without this PR<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39003">issues.joomla.org/tracker/joomla-cms/39003</a>.</sub>

File: plugins/content/loadmodule/loadmodule.php [hunk 11]
```diff
@@ -106,22 +106,24 @@ public function onContentPrepare($context, &$article, &$params, $page = 0)
...
+                $title = '';
+
+                if (array_key_exists(1, $matchesmodlist)) {
+                    $title = htmlspecialchars_decode(trim($matchesmodlist[1]));
                 }
 
-                $module = trim($matchesmodlist[0]);
-                $name   = htmlspecialchars_decode(trim($matchesmodlist[1]));
-                $stylemod  = trim($matchesmodlist[2]);
+                // Third paramater is the module style, (fallback is the plugin default set earlier).
```
Comment (brianteeman) [inline plugins/content/loadmodule/loadmodule.php]:
```suggestion
                // Third parameter is the module style, (fallback is the plugin default set earlier).
```

spelling

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 935c0ec0289c4895ec63b03aa0a906603e5c9d8f<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39003">issues.joomla.org/tracker/joomla-cms/39003</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39003">issues.joomla.org/tracker/joomla-cms/39003</a>.</sub>

**Comment by fancyFranci** [comment]
Looks good. Thank you!

**Pull request merged**